lib/node_modules/@stdlib/assert/has-same-constructor/lib/main.js

```diff
@@ -0,0 +1,53 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+/**
+* Test whether two provided values have the same constructor.
+*
+* @param {*} a - The first value to test.
+* @param {*} b - The second value to test.
+* @returns {boolean} Returns `true` if both values have the same constructor, else `false`.
+*
+* @example
+* var obj1 = new Date();
+* var obj2 = new Date();
+* console.log(hasSameConstructor(obj1, obj2));
+* // returns true, both are instances of Date
+*
+* @example
+* var obj1 = new Date();
+* var obj3 = new String("Hello");
+* console.log(hasSameConstructor(obj1, obj3));
+* // returns false
+*
+* @example
+* var obj3 = new String("Hello");
+* var obj4 = "World";
+* console.log(hasSameConstructor(obj3, obj4));
+* // returns true, one is String, the other is primitive string; but has same String constructor
+*/
+
+function hasSameConstructor(a, b) {
+    return a && b && a.constructor === b.constructor;
+}
+
+module.exports = hasSameConstructor;
```

Comment:
Before committing code, please make sure you have read the development guide and installed EditorConfig. All files should have trailing newlines.